### PR TITLE
Port to kube 1.8 and use k8s.io/code-generator

### PR DESCRIPTION
**Pull request by sttts**
Based on https://github.com/kubernetes/kubernetes/pull/52186's lib/codegen.sh. It should soon merge into kube master and will be cherry-picked to the 1.8 branch.

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -6,16 +6,16 @@ ignore:
   - net/http
   - net/textproto
 import:
-- package: k8s.io/apimachinery
-  version: release-1.6
+#- package: k8s.io/apimachinery
+#  version: release-1.7
+#- package: k8s.io/apiserver
+#  version: a9bfd91dd998b4ab203bc0bb011d4b5ac15dbaab # that's 1.7, branch is missing
```
Comment (mrIncompetent) [inline glide.yaml]: I'm not a big fan of commented out code. Can we remove this?

**Review by mrIncompetent — CHANGES_REQUESTED** [review]

File: glide.yaml (hunk 1 above)
Comment (sttts) [reply glide.yaml]: Waiting for Godo^Whttps://github.com/kubernetes/test-infra/pull/3559

File: hack/update-codegen.sh [hunk 2]
```diff
@@ -17,33 +17,35 @@
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 PACKAGE_BASE=${PACKAGE_BASE:-"github.com/kube-node/nodeset"}
 CLIENT_PATH=pkg/client
-CLIENT_NAME=clientset_v1alpha1
+CLIENT_NAME=clientset
```
Comment (ledzep2) [inline hack/update-codegen.sh]: Are we going for an unversioned client? The types are still versioned.

File: pkg/client/clientset/typed/nodeset/v1alpha1/nodeclass.go [hunk 3]
```diff
@@ -56,11 +56,42 @@ func newNodeClasses(c *NodesetV1alpha1Client) *nodeClasses {
...
+		Resource("nodeclasses").
+		VersionedParams(&opts, scheme.ParameterCodec).
+		Watch()
+}
+
 // Create takes the representation of a nodeClass and creates it.  Returns the server's representation of the nodeClass, and an error, if there is any.
 func (c *nodeClasses) Create(nodeClass *v1alpha1.NodeClass) (result *v1alpha1.NodeClass, err error) {
 	result = &v1alpha1.NodeClass{}
 	err = c.client.Post().
-		Namespace(v1alpha1.TPRNamespace).
```
Comment (ledzep2) [inline pkg/client/clientset/typed/nodeset/v1alpha1/nodeclass.go]: Are we moving to CRD only?

File: hack/update-codegen.sh (hunk 2 above)
Comment (sttts) [reply hack/update-codegen.sh]: No, still versioned. But if we get other versions as well, they will be part of the same clientset. At least that was the plan.

File: pkg/client/clientset/typed/nodeset/v1alpha1/nodeclass.go (hunk 3 above)
Comment (sttts) [reply pkg/client/clientset/typed/nodeset/v1alpha1/nodeclass.go]: TPRs will not exist anymore in 1.8, and are already deprecated in 1.7. We have to.

**Review by mrIncompetent — APPROVED** [review]

**Comment by mrIncompetent** [comment]
@ledzep2 @adieu PTAL

**Review by mrIncompetent — APPROVED** [review]

**Comment by sttts** [comment]
Merging.

**Pull request merged**